Facebook - It. 1

rodrigofegui opened the pull request:
Criado: Teste e funcionalidades para:
- Modelo de Contas do Facebook manipulados;
- Controladora e rotas para: 
  - Lista com todos as contas registradas em banco;
  - Evolução de _likes_ para uma dada conta;

rodrigofegui committed:
Add: modelo inicial dos dados manipulados do Facebook; Contém observações, potencialmente relevantes, anotadas; Add: gitIgnore -> credentials.json, no seu novo local

pedrohpe committed:
Add: modelo para acompanhamento histórico das contas do facebook

pedrohpe committed:
Add: facebook.route, facebook.controller, facebook.model; Acesso inicial à página dos dados do facebook

Co-authored-by: rodrigofegui <rodrigofegui@gmail.com>

pedrohpe committed:
Esqueleto do acesso confirmado a banco de dados com mongoDB

Co-authored-by: rodrigofegui <rodrigofegui@gmail.com>

rodrigofegui committed:
Criada a rota para a inserção de novos dados no banco de dados

Co-authored-by: pedrohpe <pedroh.egler@hotmail.com>

rodrigofegui committed:
Add: teste de acesso ao facebook;
Está:
a) realizando uma busca simples ao banco, sem manipular os resultados,
b) realizando uma inserção padrão de teste ao banco, sem, ainda, conferir a inserção, depende do processo (a),
c) mudança nos arquivos de testes existentes, resolvendo o conflito de acesso ao servidor aberto por outro teste e impossibilitanto outros

rodrigofegui committed:
Add: esqueleto de acesso aos dados locais, para povoamento do BD;

Co-authored-by: pedrohpe <pedro.egler@hotmail.com>

rodrigofegui committed:
Add: comentários em todos os métodos desenvolvidos até o momento;
Update: busca de todos os registros, exibindo-os formatados; teste de inserção para um registro único; atualização de registros já existentes.

Co-authored-by: pedrohpe <pedro.egler@hotmail.com>

rodrigofegui committed:
Add: stub para teste no BD;
Edit: atualizações necessárias para a manutenção do padrão de desenvolvimento;
Up: gerando gráfico básico.

Co-authored-by: pedrohpe = <pedroh.egler@hotmail.com>

mikaelmello commented on .gitignore:
```diff
@@ -1,6 +1,6 @@
 .DS_Store
 /node_modules
-credentials.json
+config/credentials.json
```
Deixar apenas credentials.json já ignora todos os arquivos no repositório com esse nome. Não precisa deixar o diretório explícito.

rodrigofegui replied:
Opa, excelente!
Desconhecia dessa vantagem.

mikaelmello commented on index.js:
```diff
@@ -14,9 +14,14 @@ mongoose.connect(mongoUri, { keepAlive: 1 });
+
 
-app.listen(config.port, () => {
+const server = app.listen(config.port, () => {
```
Ao invés de usar uma variável pra fechar o servidor no fim dos testes, é melhor identificar quando está sendo testado e nem fazer o bind para alguma porta. Como aqui: https://github.com/unb-cic-esw/data-viz/blob/7ac6550204fd86c9317fdb24e353fd6724a20323/index.js#L23

rodrigofegui replied:
Muito melhor assim!

mikaelmello commented on index.js:
```diff
@@ -14,9 +14,14 @@ mongoose.connect(mongoUri, { keepAlive: 1 });
 });
 
 module.exports = app;
+module.exports.server = server;
```
Aí remove isso e nem usamos server

mikaelmello commented on package.json:
```diff
@@ -3,7 +3,7 @@
   "main": "index.js",
   "scripts": {
-    "test": "jest 'server/tests' --forceExit",
+    "test": "jest 'server/tests' --forceExit --runInBand",
```
Fazendo do jeito que disse, sem fazer o bind do server para uma porta, não precisamos nem do `--forceExit` nem `do --runInBand`

mikaelmello commented on server/controllers/facebook.controller.js:
```diff
@@ -0,0 +1,284 @@
+	return facebookController;
+}; */
+
+// const mongoose = require("mongoose");
```
Aí é frescura minha, mas acho melhor apagar esses comentários acima

mikaelmello commented on server/controllers/facebook.controller.js:
```diff
@@ -0,0 +1,284 @@
+const httpStatus = require("http-status");
+const ChartNode = require("chartjs-node");
+
+if (global.CanvasGradient === undefined) {
```
Como isso é um problema para todas as branches, é melhor deixar isso definido em `index.js`

mikaelmello commented on server/routes/facebook.route.js:
```diff
@@ -0,0 +1,41 @@
+ * @param {object} req - standard request object from the Express library
+ * @param {object} res - standard response object from the Express library
+ *
+router.get("/", (req, res) => {
```
É melhor apagar comentários de rotas que não vão ser usadas (se não forem)

mikaelmello commented on server/tests/facebook.test.js:
```diff
@@ -0,0 +1,54 @@
+const request = require("supertest");
+const httpStatus = require("http-status");
+const app = require("../../index");
+const server = require("../../index").server;
```
Mesma mudança indicada no `index.js`, é melhor não usar `server`.

mikaelmello commented on server/tests/facebook.test.js:
```diff
@@ -0,0 +1,54 @@
+ */
+afterAll(async (done) => {
+	await facebookAccount.collection.drop();
+	server.close();
```
Mesma mudança indicada no `index.js`, é melhor não usar `server`.

mikaelmello commented on server/tests/spreadsheets.test.js:
```diff
@@ -3,6 +3,15 @@ const request = require("supertest");
+ * Possibility of all tests accessing the same server
+ */
+afterAll(async (done) => {
+	server.close();
```
Mesma mudança indicada no `index.js`, é melhor não usar `server`.

mikaelmello reviewed (dismissed):
Algumas frescuras e pedido para parar de usar o server e parar de fazer o servidor ouvir em alguma porta quando em ambiente de testes.

mikaelmello committed:
Resolve questões levantadas na review de #3

Principalmente parar de fazer o servidor ouvir em uma porta
quando em ambiente de testes
